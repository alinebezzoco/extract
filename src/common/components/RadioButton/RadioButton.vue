<template>
    <input type="radio" :id="id" :name="name" :value="value" v-bind="$attrs" @change="selectedItem" :disabled="disabled" />
    <label :for="id">{{ label }} </label>
</template>

<script>
export default {
    name: 'RadioButtonComponent',
    props: {
        id: {
            type: String,
            required: false
        },
        label: {
            type: String,
            required: false
        },
        value: {
            type: String,
            required: false
        },
        name: {
            type: String,
            required: false
        },
        disabled: {
            type: Boolean,
            required: false,
            default: false
        },
        isChecked: {
            type: Boolean
        },
    },
    methods: {
        selectedItem() {
            this.$emit('checked', !this.isChecked);
        }
    },
};
</script>

<style lang="scss" scoped>
@import './RadioButton.scss';
</style>