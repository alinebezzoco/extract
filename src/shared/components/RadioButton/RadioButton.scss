.radio-group {
  width: 80%;
  list-style-type: none;
  padding: 0;
  @media (max-width: 780px) {
    margin-bottom: 20px;
    width: 100%;
  }
  span {
    display: inline-block;
    position: relative;
    margin-right: 40px;
    padding: 16px 40px;
    @media (max-width: 780px) {
      margin: 0 10px;
      padding: 16px 32px;
    }
  }
  label,
  input {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
  input[type="radio"] {
    opacity: 0.01;
    z-index: 100;
    &:checked + label {
      background: var(--primary-color);
      border-radius: 32px;
      color: var(--white);
    }
    + label:hover {
      background: var(--light-blue);
      color: var(--primary-color);
      border-radius: 32px;
    }
  }
  label {
    padding: 5px;
    cursor: pointer;
    z-index: 90;
    line-height: 22px;
    font-size: 16px;
    color: var(--primary-color);
    @media (max-width: 780px) {
      font-size: 14px;
    }
  }
}
