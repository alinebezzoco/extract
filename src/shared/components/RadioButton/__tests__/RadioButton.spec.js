import { mount } from '@vue/test-utils';
import { render, fireEvent, screen } from '@testing-library/vue';
import RadioButton from "../RadioButton.vue";

const mockProps = {
  label: 'My Label'
};
const mockAttrs = { id: 'id' };
const mockIdSelector = '#id';
const mockLabel = 'label';

describe('RadioButton.vue', () => {
  it('should render the component', () => {
    const container = mount(RadioButton);

    expect(container.exists()).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should change value of radio button by clicking label', async () => {
    const container = mount(RadioButton, {
      propsData: mockProps,
      attrs: mockAttrs
    });
    const label = container.find(mockLabel);

    await label.trigger('click');
    const input = container.find(mockIdSelector);

    expect((input.element).checked).toBeTruthy();
  });

  it('should change value when radio button checked is true', () => {
    render(RadioButton);
    const radio = screen.getByRole('radio');

    fireEvent.change(radio, { target: { checked: true } });

    expect(radio).toHaveProperty('checked', true);
  });

  it('should change value when radio button checked is false', () => {
    render(RadioButton);
    const radio = screen.getByRole('radio');

    fireEvent.change(radio, { target: { checked: false } });

    expect(radio).toHaveProperty('checked', false);
  });
});
