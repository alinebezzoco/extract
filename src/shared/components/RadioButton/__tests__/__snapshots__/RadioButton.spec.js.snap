// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton.vue should render the component 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <span>
        <input
          type="radio"
        />
        <label />
      </span>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "disabled": false,
          "hasOwnProperty": [Function],
          "id": undefined,
          "isChecked": false,
          "label": undefined,
          "name": undefined,
          "selectedItem": [Function],
          "value": undefined,
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "disabled": 3,
            "hasOwnProperty": 4,
            "id": 3,
            "isChecked": 3,
            "isVueInstance": 0,
            "label": 3,
            "name": 3,
            "selectedItem": 4,
            "toJSON": 0,
            "value": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "disabled": false,
            "hasOwnProperty": [Function],
            "id": undefined,
            "isChecked": false,
            "label": undefined,
            "name": undefined,
            "selectedItem": [Function],
            "value": undefined,
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "disabled": false,
            "id": undefined,
            "isChecked": false,
            "label": undefined,
            "name": undefined,
            "value": undefined,
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "disabled": Object {
                "0": true,
                "1": true,
                "default": false,
                "required": false,
                "type": [Function],
              },
              "id": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "isChecked": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
              "label": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "name": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "value": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
            },
            Array [
              "disabled",
              "isChecked",
            ],
          ],
          "provides": Object {},
          "proxy": Object {
            "disabled": false,
            "hasOwnProperty": [Function],
            "id": undefined,
            "isChecked": false,
            "label": undefined,
            "name": undefined,
            "selectedItem": [Function],
            "value": undefined,
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "id",
                  "name",
                  "value",
                  "disabled",
                ],
                "el": <input
                  type="radio"
                />,
                "key": null,
                "patchFlag": 16,
                "props": Object {
                  "disabled": false,
                  "id": undefined,
                  "name": undefined,
                  "onChange": [Function],
                  "type": "radio",
                  "value": undefined,
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "for",
                ],
                "el": <label />,
                "key": null,
                "patchFlag": 9,
                "props": Object {
                  "for": undefined,
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "id",
                  "name",
                  "value",
                  "disabled",
                ],
                "el": <input
                  type="radio"
                />,
                "key": null,
                "patchFlag": 16,
                "props": Object {
                  "disabled": false,
                  "id": undefined,
                  "name": undefined,
                  "onChange": [Function],
                  "type": "radio",
                  "value": undefined,
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "for",
                ],
                "el": <label />,
                "key": null,
                "patchFlag": 9,
                "props": Object {
                  "for": undefined,
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
            ],
            "dynamicProps": null,
            "el": <span>
              <input
                type="radio"
              />
              <label />
            </span>,
            "key": null,
            "patchFlag": 0,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "span",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "methods": Object {
              "selectedItem": [Function],
            },
            "name": "RadioButtonComponent",
            "props": Object {
              "disabled": Object {
                "0": true,
                "1": true,
                "default": false,
                "required": false,
                "type": [Function],
              },
              "id": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "isChecked": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
              "label": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "name": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
              "value": Object {
                "0": false,
                "1": true,
                "required": false,
                "type": [Function],
              },
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <span>
          <input
            type="radio"
          />
          <label />
        </span>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "methods": Object {
            "selectedItem": [Function],
          },
          "name": "RadioButtonComponent",
          "props": Object {
            "disabled": Object {
              "0": true,
              "1": true,
              "default": false,
              "required": false,
              "type": [Function],
            },
            "id": Object {
              "0": false,
              "1": true,
              "required": false,
              "type": [Function],
            },
            "isChecked": Object {
              "0": true,
              "1": true,
              "type": [Function],
            },
            "label": Object {
              "0": false,
              "1": true,
              "required": false,
              "type": [Function],
            },
            "name": Object {
              "0": false,
              "1": true,
              "required": false,
              "type": [Function],
            },
            "value": Object {
              "0": false,
              "1": true,
              "required": false,
              "type": [Function],
            },
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <span>
          <input
            type="radio"
          />
          <label />
        </span>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "disabled": false,
    "hasOwnProperty": [Function],
    "id": undefined,
    "isChecked": false,
    "label": undefined,
    "name": undefined,
    "selectedItem": [Function],
    "value": undefined,
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": <span>
    <input
      type="radio"
    />
    <label />
  </span>,
}
`;
